mlir/include/mlir/Dialect/SPIRV/IR/SPIRVAtomicOps.td

```diff
@@ -14,8 +14,20 @@
 #ifndef MLIR_DIALECT_SPIRV_IR_ATOMIC_OPS
 #define MLIR_DIALECT_SPIRV_IR_ATOMIC_OPS
 
-class SPIRV_AtomicUpdateOp<string mnemonic, list<Trait> traits = []> :
-  SPIRV_Op<mnemonic, traits> {
+include "mlir/Dialect/SPIRV/IR/SPIRVBase.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+class PointeeTypeMatchTrait<string pointer, string name>
+    : TypesMatchWith<
+      "$" # name # " type matches pointee type of " # "$" # pointer, pointer,
```

Comment:
This doesn't perform any name substitution with '$'. Was is supposed to? We should either make it print the types when possible or remove the dollar signs.